pkg/csi_driver/gcfs_driver.go

```diff
@@ -267,3 +308,130 @@ func (driver *GCFSDriver) Run(endpoint string) {
 	}
 	s.Wait()
 }
+
+func runMultishareReconciler(driverConfig *GCFSDriverConfig) {
+	config, err := util.BuildConfig(driverConfig.FeatureOptions.FeatureStateful.KubeConfig)
+	if err != nil {
+		klog.Error(err.Error())
+		os.Exit(1)
+	}
+	config.QPS = (float32)(driverConfig.FeatureOptions.FeatureStateful.KubeAPIQPS)
+	config.Burst = driverConfig.FeatureOptions.FeatureStateful.KubeAPIBurst
+
+	kubeClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		klog.Error(err.Error())
+		os.Exit(1)
+	}
+	fsClient, err := clientset.NewForConfig(config)
+	if err != nil {
+		klog.Error(err.Error())
+		os.Exit(1)
+	}
+	driverfsClient, err := clientset.NewForConfig(config)
+	if err != nil {
+		klog.Error(err.Error())
+		os.Exit(1)
+	}
+
+	resyncPeriod := driverConfig.FeatureOptions.FeatureStateful.ResyncPeriod
+	factory := fsInformers.NewSharedInformerFactory(fsClient, resyncPeriod)
+	coreFactory := informers.NewSharedInformerFactory(kubeClient, resyncPeriod)
+	driverFactory := fsInformers.NewSharedInformerFactory(driverfsClient, resyncPeriod)
+	sharescheme.AddToScheme(scheme.Scheme)
+
+	recon := NewMultishareReconciler(
+		fsClient,
+		driverConfig,
+		factory.Multishare().V1alpha1().ShareInfos(),
+		factory.Multishare().V1alpha1().InstanceInfos(),
+		coreFactory.Storage().V1().StorageClasses().Lister(),
+	)
+	driverConfig.Reconciler = recon
+	driverConfig.FeatureOptions.FeatureStateful.DriverClientSet = driverfsClient
+	driverConfig.FeatureOptions.FeatureStateful.ShareLister = driverFactory.Multishare().V1alpha1().ShareInfos().Lister()
+
+	if err := ensureCustomResourceDefinitionsExist(fsClient); err != nil {
+		klog.Errorf("Exiting due to failure to ensure CRDs exist during startup: %+v", err)
+		os.Exit(1)
+	}
+
+	run := func(context.Context) {
+		// run...
+		stopCh := make(chan struct{})
+		factory.Start(stopCh)
+		coreFactory.Start(stopCh)
+		driverFactory.Start(stopCh)
+		go recon.Run(stopCh)
+
+		// ...until SIGINT
+		c := make(chan os.Signal, 1)
+		signal.Notify(c, os.Interrupt)
+		<-c
+		close(stopCh)
+	}
+
+	statefulConfig := driverConfig.FeatureOptions.FeatureStateful
+
+	if !statefulConfig.LeaderElection {
+		go run(context.TODO())
+	} else {
+		go func() {
+			lockName := "filestore-stateful-leader"
+			config, err := util.BuildConfig(driverConfig.FeatureOptions.FeatureStateful.KubeConfig)
+			if err != nil {
+				klog.Fatalf(err.Error())
+			}
+
+			leClient, err := kubernetes.NewForConfig(config)
+			if err != nil {
+				klog.Fatalf("failed to create leaderelection client: %v", err)
```

Comment:
Full displayed messages should be capitalized 
go/go-style/decisions#error-strings

Reply:
done